src/cli/cli_tcp.cpp

```diff
@@ -194,27 +229,36 @@ otError TcpExample::ProcessSend(Arg aArgs[])
     otError error;
 
     VerifyOrExit(mInitialized, error = OT_ERROR_INVALID_STATE);
-    VerifyOrExit(!mSendBusy, error = OT_ERROR_BUSY);
     VerifyOrExit(mBenchmarkBytesTotal == 0, error = OT_ERROR_BUSY);
 
-    mSendLink.mNext = nullptr;
-    mSendLink.mData = mSendBuffer;
-    VerifyOrExit(!aArgs[0].IsEmpty(), error = OT_ERROR_INVALID_ARGS);
-    mSendLink.mLength = OT_MIN(aArgs[0].GetLength(), sizeof(mSendBuffer));
-    memcpy(mSendBuffer, aArgs[0].GetCString(), mSendLink.mLength);
-    VerifyOrExit(aArgs[1].IsEmpty(), error = OT_ERROR_INVALID_ARGS);
+    if (mUseCircularSendBuffer)
+    {
+        size_t written;
```

Comment:
It's not needed in this case. We need `mSendBusy` for linked buffers because we only have one `mSendLink` and we have to make sure it's not in use. For the circular buffer, the write call will simply return short if the circular buffer fills up.